src/bot/commands/userinfo.js

```diff
@@ -1,69 +1,57 @@
-      if (member.permissions.json[perm] === true && notablePermissions.indexOf(perm) !== -1) {
-        perms.push(perm)
-      }
-    })
-    if (perms.length === 0) {
-      perms.push('None')
-    }
-    fields.push({
-      name: 'Name',
-      value: `**${member.username}#${member.discriminator}** ${member.nick ? `(**${member.nick}**)` : ''} (${member.id})`
-    }, {
-      name: 'Join Date',
-      value: `**${new Date(member.joinedAt)}** (${Math.round((new Date().getTime() - member.joinedAt) / (1000 * 60 * 60 * 24))} days)`
-    }, {
-      name: 'Creation Date',
-      value: `**${new Date(member.createdAt).toUTCString()}**`
-    }, {
-      name: 'Roles',
-      value: `${member.roles.length !== 0 ? member.roles.map(r => `\`${message.channel.guild.roles.get(r).name}\``).join(', ') : 'None'}`
-    }, {
-      name: 'Notable Permissions',
-      value: `\`${perms.join(', ')}\``
-    })
-    message.channel.createMessage({
-      embed: {
-        timestamp: new Date(message.timestamp),
-        color: color,
-        thumbnail: {
-          url: member.avatar ? member.avatarURL : `https://cdn.discordapp.com/embed/avatars/${member.discriminator % 5}.png`
-        },
-        fields: fields
-      }
-    }).catch(() => { })
-  },
-  name: 'userinfo',
-  description: 'Use this with a mention to get info about a user.', // The restriction of using a mention is very intentional.
-  type: 'any',
-  category: 'General'
-}
+    func: async message => {
+      let member = message.member;
+      if (message.mentions.length !== 0) member = message.channel.guild.members.get(message.mentions[0].id)
+      const fields = []
+      const perms = []
+      Object.keys(member.permissions.json).forEach((perm) => {
+        if (member.permissions.json[perm] === true && notablePermissions.indexOf(perm) !== -1) {
+          perms.push(`\`${perm}\``)
+        }
+      })
+      const roles = member.roles.map(r => message.channel.guild.roles.get(r)).sort((a,b)=>b.position-a.position);
+      fields.push({
+        name: 'Name',
+        value: `**${member.username}#${member.discriminator}** ${member.nick ? `(**${member.nick}**)` : ''} (${member.id})`
+      }, {
+        name: 'Join Date',
+        value: `**${new Date(member.joinedAt).toUTCString()}** (${Math.round((new Date().getTime() - member.joinedAt) / (1000 * 60 * 60 * 24))} days)`
+      }, {
+        name: 'Creation Date',
+        value: `**${new Date(member.createdAt).toUTCString()}** (${Math.round((new Date().getTime() - member.createdAt) / (1000 * 60 * 60 * 24))} days)`
+      }, {
+        name: 'Roles',
+        value: roles.length !== 0 ? roles.map(c => `\`${c.name}\``).join(", ") : 'None'
+      }, {
+        name: 'Notable Permissions',
+        value: perms.length !== 0 ? perms.join(", ") : 'None'
+      })
+      message.channel.createMessage({
+        embed: {
+          timestamp: new Date(message.timestamp),
+          color: roles[0].color,
```

Comment:
If the user has no roles, this errors. Using the highest role color for the embed color is fine, but you may need a ternary or logic above to set this correctly.

Reply:
Sorted